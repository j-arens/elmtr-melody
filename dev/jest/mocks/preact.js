module.exports = {
    unmountComponent: jest.fn(),
};
