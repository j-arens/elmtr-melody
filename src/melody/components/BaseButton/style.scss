.btn {
    border: none;
    background: transparent;
    cursor: pointer;
    padding: 0;
    margin: 0;
    line-height: initial;

    &:hover {
        background: transparent;
    }
    
    &:focus {
        outline: 0;
        background: transparent;
    }
    
    &:active {
        transform: scale(0.95);
        background: transparent;
    }
}