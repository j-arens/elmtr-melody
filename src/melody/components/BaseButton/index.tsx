import { WithOptionalClassName } from '@components/type';
import { h } from 'preact';
const s = require('./style.scss');

interface Props extends WithOptionalClassName {
    children?: JSX.Element[];
    onClick?: (e: Event) => any;
    key?: string;
}

export default (props: Props) => (
    <button
        {...props}
        class={`${s.btn} ${props.className || ''}`}
        role="button"
    >
        {props.children}
    </button>
);
