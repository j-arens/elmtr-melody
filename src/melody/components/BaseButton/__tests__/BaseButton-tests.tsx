// import render from 'preact-render-to-string';
const render = require('preact-render-to-string');
import { h } from 'preact';
import BaseButton from '../';

describe('SimpleToolbar', () => {
    it('renders', () => {
        const result = render(<BaseButton>hi</BaseButton>);
        expect(result).toMatchSnapshot();
    });
});
