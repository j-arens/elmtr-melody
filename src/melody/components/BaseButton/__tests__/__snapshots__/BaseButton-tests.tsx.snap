// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleToolbar renders 1`] = `<button class="btn ">hi</button>`;
